src/pages/newpage.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import * as actionCreators from '../redux/action_creators'
+
+
+
+class NewPage extends Component {
+    constructor(props) {
+        super(props)
+
+        this.setGreeting = this.setGreeting.bind(this)
+
+        this.state = {
+            greeting: 'Hello Olio',
+        }
+    }
+
+    static get contextTypes() {
+        return { router: React.PropTypes.object.isRequired }
+    }
+
+    setGreeting(e) {
+        this.setState({ greeting: e.target.value })
+    }
+
+
+    render() {
+        return (
+            <div>
+                <h3>Enterer New Greeting: </h3>
+
+                <input type="text" onChange={this.setGreeting} />
```

Comment:
consider adding  `value=this.state.greeting` this will tie the value of the text input to the value in state. This is generally the preferred way of doing react inputs.